nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentModelJsonWriter.java

```diff
@@ -258,4 +264,23 @@ private void writeSchemaProperties(JsonGenerator jg, DocumentModel doc, String s
         }
     }
 
+    protected void withDocumentAttached(DocumentModel doc, ThrowableRunnable<IOException> runnable) throws IOException {
+        String repositoryName = doc.getRepositoryName();
+        CoreSession currentSession = ctx.getSession(null).getSession();
+        CloseableCoreSession session = null;
+        try {
+            if (!currentSession.getRepositoryName().equals(repositoryName)) {
+                session = CoreInstance.openCoreSession(repositoryName);
+                // make sure we can re-attach the document
+                doc.detach(false);
+                doc.attach(session.getSessionId());
+            }
+            runnable.run();
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
```

Comment:
Suggestion:

          boolean sameRepository = currentSession.getRepositoryName().equals(repositoryName);
          try (CloseableCoreSession session = sameRepository ? null :  CoreInstance.openCoreSession(repositoryName)) {
              if (session != null) {
                  // re-attach the document
                  doc.detach(false);
                  doc.attach(session.getSessionId());
              }
              runnable.run();
          }